File: tp-arithmetic-java/src/main/java/fr/isima/controller/ArithmeticController.java
package fr.isima.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import java.text.NumberFormat;

@Controller
@RequestMapping("/maths")
public class ArithmeticController {

    @GetMapping("/diviser")
    @ResponseBody
    public String divide(@RequestParam("a") double a, @RequestParam("b") double b) {
        return format(a / b);
    }

    private String format(double v) {
        final NumberFormat number = NumberFormat.getInstance();
        number.setMaximumFractionDigits(4);
        return number.format(v);
    }

    @GetMapping("/soustraire")
    @ResponseBody
    public String soustraire(@RequestParam("a") double a, @RequestParam("b") double b) {
        return format(a - b);
    }

    @GetMapping("/additionner")
    @ResponseBody
    public String additionner(@RequestParam("a") double a, @RequestParam("b") double b) {
        return format(a + b);
    }

    @GetMapping("/multiplier")
    @ResponseBody
    public String multiplier(@RequestParam("a") double a, @RequestParam("b") double b) {
        return format(a * b);
    }
}

File: tp-arithmetic-kotlin/src/main/kotlin/fr/isima/controller/ArithmeticController.kt
package fr.isima.controller

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.ResponseBody

import java.text.NumberFormat

@Controller
@RequestMapping("/maths")
class ArithmeticController {

    @GetMapping("/diviser")
    @ResponseBody
    fun divide(@RequestParam("a") a: Double, @RequestParam("b") b: Double) = (a / b).format()

    @GetMapping("/soustraire")
    @ResponseBody
    fun soustraire(@RequestParam("a") a: Double, @RequestParam("b") b: Double) = (a - b).format()

    @GetMapping("/additionner")
    @ResponseBody
    fun additionner(@RequestParam("a") a: Double, @RequestParam("b") b: Double) =  (a + b).format()

    @GetMapping("/multiplier")
    @ResponseBody
    fun multiplier(@RequestParam("a") a: Double, @RequestParam("b") b: Double) = (a * b).format()
}

private fun Double.format() = NumberFormat.getInstance().apply { maximumFractionDigits = 4 }.run { format(this@format) }